src/customActions/configurationParse.ts

```diff
@@ -1,30 +1,29 @@
-import * as fs from "async-file";
-import { customActions } from "./customActions";
-import { CustomActionInfo } from "../model/configurationInfo";
-import { getRealPath } from "../utils/path.utils";
-import * as vscode from "vscode";
+import * as fs from 'async-file';
+import * as filePath from 'path';
+import { customActions } from './customActions';
+import { CustomActionInfo } from '../model/configurationInfo';
+import { getRealPath } from '../utils/path.utils';
+import * as vscode from 'vscode';
 
 class ConfigurationParse {
-  private async isWhenExist(path: string, whenPath: string): Promise<boolean> {
+  private async isWhenExist(path: string | undefined, whenPath: string): Promise<boolean> {
     try {
-      if (whenPath && (await fs.exists(path + "\\" + whenPath))) {
+      if (path && whenPath && (await fs.exists(filePath.join(path, whenPath)))) {
         return true;
       }
     } catch (e) {
-      console.log("TerminalStatusBar");
+      console.log('TerminalStatusBar');
       console.log(e);
     }
     return false;
   }
 
   public async parse(fullPath: string) {
-    const fileContent = await fs.readTextFile(fullPath);
     try {
+      const fileContent = await fs.readTextFile(fullPath);
       this.create(JSON.parse(fileContent));
     } catch {
-      vscode.window.showInformationMessage(
-        "Terminal Status bar: Problem with parsing the json"
-      );
+      vscode.window.showInformationMessage('Terminal Status bar: Problem with parsing the json');

```

Comment:
Syntax errors are very common, and they should be specific message about it. If we add "readTextFile(fullPath);"  inside the try, this is not true anymore. Because if the "readtextfile" collapses, the message will appear for the wrong reasons.

Reply:
Cool, in that case it's more likely to be better to be more verbose about where the error is happening so also split up the create fn call, parsing the json and reading the file. :+1: